framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -119,236 +86,158 @@ private static Pattern anchored(String regex) {
+                    && !context.parsingMember) {
+                return parseParameter(expr.asNameExpr().getNameAsString(), context);
+            } else if (expr.isNameExpr()) {
+                return parseIdentifier(expr.asNameExpr().getNameAsString(), env, path, context);
+            } else if (expr.isMethodCallExpr()) {
+                return parseMethodCall(expr.asMethodCallExpr(), context, path, env);
+            } else if (expr.isFieldAccessExpr()) {
+                return parseFieldAccess(expr.asFieldAccessExpr(), env, context, path);
+            } else if (expr.isClassExpr() && !context.parsingMember) {
+                return parseClassExpression(expr.asClassExpr(), env, context, path);
             } else {
-                return null;
+                String message;
+                if (expression.equals("#0")) {
+                    message =
+                            "one should use \"this\" for the receiver or \"#1\" for the first formal parameter";
+                } else {
+                    message = "is an unrecognized expression";
+                }
+                if (context.parsingMember) {
+                    message += " in a context with parsingMember=true";
+                }
+                throw constructParserException(expression, message);
             }
         }
 
-        int i = s.indexOf(".");
-        if (i == -1) {
-            return null;
-        }
+        throw constructParserException(expression, "is an unrecognized expression");
+    }
 
-        String receiver = s.substring(0, i);
-        String remaining = s.substring(i + 1);
+    /** Replaces every occurrence of "#(number)" with "_param_(number)" */
+    private static String noHashTags(String expression) {
+        String updatedExpression = expression;
 
-        return Pair.of(receiver, remaining);
+        for (Integer integer : parameterIndices(expression)) {
+            updatedExpression = updatedExpression.replaceAll("#" + integer, "_param_" + integer);
+        }
+
+        return updatedExpression;
     }
 
-    private static Receiver parseMemberSelect(
-            String s, ProcessingEnvironment env, FlowExpressionContext context, TreePath path)
+    private static Receiver parseFieldAccess(
+            FieldAccessExpr expr,
+            ProcessingEnvironment env,
+            FlowExpressionContext context,
+            TreePath path)
             throws FlowExpressionParseException {
-        Pair<String, String> select = parseMemberSelect(s);
-        assert select != null : "isMemberSelect must be called first";
-
         Receiver receiver;
         String memberSelected;
 
         Resolver resolver = new Resolver(env);
 
         // Attempt to match a package and class name first.
         Pair<ClassName, String> classAndRemainingString =
-                matchPackageAndClassNameWithinExpression(s, resolver, path);
+                matchPackageAndClassNameWithinExpression(expr.toString(), resolver, path);
```

Comment:
Do you still need this call? Or would it be sufficient to just use the `else` branch of the `if` statement below and remove this method?